### PR TITLE
Feature: Allow domain suggestions containing a dot by using query parameter instead of a path segment

### PR DESCRIPTION
**Pull request by adamziel**
I would like to be able to link to a domain suggestion page and suggest domain name containing dots. Currently it is not possible, because the URL would look like this: `/domains/add/suggestion/mytestsite.com/mytestsite.wordpress.com` and `getSiteFragment()` will think that site fragment is `mytestsite.com`. I issued a PR to fix it here:

https://github.com/Automattic/wp-calypso/pull/22493

But it's not getting a lot of thumbs up since the entire calypso would be affected and we don't really have tests to cover all of that :-) So I prepared this PR that touches just the parts I need to update.

As a side note, I wanted to prepare some tests cases, but we cannot really test `controller.jsx`. It imports some modules expecting `documentElement` to be defined and we don't have any shim for that in our test environment.

### CODE REVIEW

File: client/my-sites/domains/controller.jsx [hunk 1]
```diff
@@ -229,6 +235,7 @@ export default {
 	googleAppsWithRegistration,
 	redirectIfNoSite,
 	redirectToAddMappingIfVipSite,
+	redirectToDomainSearchWithSuggestionAsQuery,
```
Comment (simison) [inline client/my-sites/domains/controller.jsx]: This is a bit mouthful — how about `redirectToDomainSearchSuggestion` ?

File: client/my-sites/domains/index.js [hunk 2]
```diff
@@ -247,7 +247,7 @@ export default function() {
 			domainsController.redirectIfNoSite( '/domains/add' ),
 			domainsController.redirectToAddMappingIfVipSite(),
 			jetPackWarning,
-			domainsController.domainSearch,
+			domainsController.redirectToDomainSearchWithSuggestionAsQuery,
 			makeLayout,
 			clientRender
```
Comment (simison) [inline client/my-sites/domains/index.js]: If the last middleware always redirects, we don't need `makeLayout` and `clientRender` here anymore. They just handle the final rendering (which now will happen in some other route).

File: client/my-sites/domains/domain-search/domain-search.jsx [hunk 3]
```diff
@@ -143,7 +143,7 @@ class DomainSearch extends Component {
 						>
 							<RegisterDomainStep
 								path={ this.props.context.path }
-								suggestion={ this.props.context.params.suggestion }
+								suggestion={ this.props.context.query.s }
```
Comment (simison) [inline client/my-sites/domains/domain-search/domain-search.jsx]: Not technically an issue but just noting here: elsewhere in Calypso `?s=` means "search" (e.g. users, pages, media...)

**Comment by adamziel** [comment]
@simison done! I'd argue that `s=` meaning `search=` would still make sense on /domains/add page since it would mean "search for this and this domain", but I renamed it to `suggestion=` because it makes sense it could save some confusion to future developers.

**Comment by adamziel** [comment]
Moved here: Feature: Allow domain suggestions containing a dot by using query parameter instead of a path segment

**Pull request closed without merging**